services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -333,6 +354,53 @@ private RegistrationResult getRegistrationResult(final String deviceId, final Js
                 });
     }
 
+    private Future<OperationResult<List<DeviceWithId>>> processSearchDevices(
+            final String tenantId,
+            final int pageSize,
+            final int pageOffset,
+            final List<Filter> filters,
+            final List<Sort> sortOptions) {
+
+        final FindOptions searchDevicesOptions = new FindOptions()
+                .setLimit(pageSize)
+                .setSkip(pageOffset);
+        final JsonObject searchDevicesQuery = MongoDbDocumentBuilder.builder()
+                .withTenantId(tenantId)
+                .withDeviceFilters(filters)
+                .document();
+        final JsonObject sortDocument = MongoDbDocumentBuilder.builder()
+                .withDeviceSortOptions(sortOptions)
+                .document();
+        final Promise<List<JsonObject>> searchDevicesPromise = Promise.promise();
+
+        searchDevicesOptions.setSort(sortDocument);
+        LOG.trace("pageSize: [{}], pageOffset: [{}], searchDevicesQuery: [{}], sortOptions: [{}]", pageSize, pageOffset,
```

Comment:
this should be put into a
```
if (LOG.isTraceEnabled()) {
}
```
block in order to prevent serialization of query and sort documents if not required

Reply:
I have incorporated it and pushed the changes.